frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/activity-card/activity-card.tsx

```diff
@@ -4,68 +4,91 @@ import DashboardCard from '@console/shared/src/components/dashboard/dashboard-ca
+import { SubscriptionKind } from '@console/operator-lifecycle-manager';
+import {
+  CEPH_STORAGE_NAMESPACE,
+  OCS_OPERATOR,
+  PVC_FILTER_ANNOTATION,
+} from '../../../../constants/index';
 import { DATA_RESILIENCY_QUERY, StorageDashboardQuery } from '../../../../constants/queries';
+import {
+  pvcResource,
+  subscriptionResource,
+  storageClusterResource,
+  eventsResource,
+} from '../../../../constants/resources';
 import { getResiliencyProgress } from '../../../../utils';
-import { OCSServiceModel } from '../../../../models';
 import { isClusterExpandActivity, ClusterExpandActivity } from './cluster-expand-activity';
 import { isOCSUpgradeActivity, OCSUpgradeActivity } from './ocs-upgrade-activity';
 import './activity-card.scss';
 
-const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
-const subscriptionResource: FirehoseResource = {
-  isList: true,
-  kind: referenceForModel(SubscriptionModel),
-  namespaced: false,
-  prop: 'subs',
-};
-
-const storageClusterResource: FirehoseResource = {
-  isList: true,
-  kind: referenceForModel(OCSServiceModel),
-  namespaced: false,
-  prop: 'storage-cluster',
-};
-
 export const getOCSSubscription = (subscriptions: FirehoseResult): SubscriptionKind => {
   const itemsData: K8sResourceKind[] = subscriptions?.data;
   return _.find(itemsData, (item) => item?.spec?.name === OCS_OPERATOR) as SubscriptionKind;
 };
 
-const ocsEventNamespaceKindFilter = (event: EventKind): boolean => {
-  const eventKind = event?.involvedObject?.kind;
-  const eventNamespace = getNamespace(event);
-  return eventNamespace === CEPH_STORAGE_NAMESPACE || eventKind === PersistentVolumeClaimModel.kind;
-};
-
 const RecentEvent = withDashboardResources(
   ({ watchK8sResource, stopWatchK8sResource, resources }: DashboardItemProps) => {
     React.useEffect(() => {
       watchK8sResource(eventsResource);
+      watchK8sResource(pvcResource);
       return () => {
         stopWatchK8sResource(eventsResource);
+        stopWatchK8sResource(pvcResource);
       };
     }, [watchK8sResource, stopWatchK8sResource]);
-    return (
+
+    const ocsEventNamespaceKindFilter = (event: EventKind): boolean => {
+      const eventKind = event?.involvedObject?.kind;
+      const eventNamespace = getNamespace(event);
+      const pvcItems = resources?.pvcs as FirehoseResult<PersistentVolumeClaimKind[]>;
+      if (eventKind === PersistentVolumeClaimModel.kind) {
+        const pvcObj = _.find(
+          pvcItems.data,
+          (obj: PersistentVolumeClaimKind) => obj?.metadata?.name === event?.involvedObject?.name,
+        );
+        const annotations = getAnnotations(pvcObj);
+        return (
+          annotations &&
+          annotations[PVC_FILTER_ANNOTATION] &&
+          isCephProvisioner(annotations[PVC_FILTER_ANNOTATION])
+        );
+      }
+      return eventNamespace === CEPH_STORAGE_NAMESPACE;
+    };
+
```

Comment:
It would be better to create a map for pvcName-annotation prior and then utilize in this filter function. Both events and pvc can go high in numbers and you will end up with n^2

Reply:
FYI the filter will show only events for existing PVCs. Its a tradeoff between this or showing events for all PVCs - non ceph ones but includes deleted ones too and only existing ceph PVCs. Is that really what you prefer ?

Reply:
Yes, thats what we want to do here as per [UX](https://issues.redhat.com/browse/RHSTOR-1106?focusedCommentId=14173091&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14173091)

Reply:
The reason being , OCS supports ceph based storage classes only. PV/PVCs created by other storage classes should be filtered out.

Reply:
@afreen23, as per @rawagner's comment, if we keep this filter, the events of deleted PVC will no longer be listed (being Ceph or non-ceph), the Ceph one as well which were appearing before will also disappear as the events won't match any PVC in pvc list, doesn't that take away the purpose of showing events?

Reply:
We sorted this offline, commenting for reference here:  Its a tradeoff but we need to have ocs based events only in the dashboard. All events can be looked anytime in console.